rust/src/transformations/make_stable_expr/expr_binary/mod.rs

```diff
@@ -62,50 +59,61 @@ where
-            series_domain
-        }
-        Plus | Minus | Multiply | Divide | TrueDivide | FloorDivide => {
-            let common_dtype = try_get_supertype(&left_series.dtype(), &right_series.dtype())?;
-
-            let out_dtype = match op {
-                Plus | Minus | Multiply | FloorDivide => common_dtype,
-                TrueDivide | Divide => {
-                    if common_dtype.is_float() {
-                        common_dtype
-                    } else {
-                        DataType::Float64
-                    }
-                }
-                _ => unreachable!("due to above match arm"),
-            };
-
-            let mut series_domain = SeriesDomain::new_from_field(Field::new(
-                left_series.name.clone(),
-                out_dtype.clone(),
-            ))?;
-
-            // output is nullable when op is FloorDivide since a // 0 is null for any a
-            series_domain.nullable =
-                left_series.nullable || right_series.nullable || matches!(op, FloorDivide);
-
-            series_domain
-        }
-        _ => {
-            return fallible!(
-                MakeTransformation,
-                "unsupported operator: {:?}. Only arithmetic operations or binary operations that emit booleans are currently supported.",
-                op
-            );
-        }
-    };
+    if !matches!(
+        op,
+        Eq | EqValidity
+            | NotEq
+            | NotEqValidity
+            | Lt
+            | LtEq
+            | Gt
+            | GtEq
+            | Plus
+            | Minus
+            | Multiply
+            | Divide
+            | TrueDivide
+            | FloorDivide
+            | Modulus
+            | And
+            | Or
+            | Xor
+            | LogicalAnd
+            | LogicalOr
+    ) {
+        return fallible!(
+            MakeTransformation,
+            "unsupported binary operator: {:?}. Please open an issue on the OpenDP GitHub if you would like to see this supported.",
+            op
+        );
+    }
+    // use Polars to compute the output dtype
+    let mock_df = DataFrame::new(vec![
+        Column::new_empty("left".into(), &left_series.dtype()),
+        Column::new_empty("right".into(), &right_series.dtype()),
+    ])?;
+    let out_dtype = mock_df
+        .lazy()
+        .select([binary_expr(col("left"), op, col("right"))])
+        .collect()?
+        .column("left")?
+        .dtype()
+        .clone();
+
+    let field = Field::new(expr_output_name(&expr)?, out_dtype.clone());
+    let mut series_domain = SeriesDomain::new_from_field(field)?;
+
+    // division by zero may introduce null values
+    series_domain.nullable = left_series.nullable
+        || right_series.nullable
+        || matches!(op, FloorDivide | TrueDivide | Divide);
+
+    if matches!(op, EqValidity | NotEqValidity) {
+        series_domain.nullable = false;
+    }
```

Comment:
Combining this into one big expression might be a little safer over time. Something like:
```
    series_domain.nullable = ! matches!(op, EqValidity | NotEqValidity) && (
        left_series.nullable
        || right_series.nullable
        || matches!(op, FloorDivide | TrueDivide | Divide));

Reply:
This is more difficult to read, IMO. 

Currently, there's some cases that could introduce nulls. And then there's some ops that eliminate all nulls.